release-notes/3.0/3.0.0-preview2.md

```diff
@@ -0,0 +1,115 @@
+- Upgraded Native SNI dependency to **v3.0.0-preview1** along with enhanced event tracing support [#1006](https://github.com/dotnet/SqlClient/pull/1006) [Read more](#event-tracing-improvements-in-sni.dll)
+
+### Fixed
+- Fixed wrong data blended with transactions in .NET Core by marking a connection with open result to be doomed [#1023](https://github.com/dotnet/SqlClient/pull/1023)
+- Fixed derived parameters containing typename incorrectly [#1020](https://github.com/dotnet/SqlClient/pull/1020)
+- Fixed server connection leak possibilities when an exception occurs in pooling layer [#890](https://github.com/dotnet/SqlClient/pull/890)
+- Fixed IP connection resolving logic in .NET Core [#1016](https://github.com/dotnet/SqlClient/pull/1016) [#1031](https://github.com/dotnet/SqlClient/pull/1031)
+
+### Changed
+- Performance improvements in `SqlDateTime` to `DateTime` internal conversion method [#912](https://github.com/dotnet/SqlClient/pull/912)
+- Improved memory allocation by avoiding unnecessary context switching [1008](https://github.com/dotnet/SqlClient/pull/1008)
+- Various performance improvements [#963](https://github.com/dotnet/SqlClient/pull/963) [#996](https://github.com/dotnet/SqlClient/pull/996) [#1004](https://github.com/dotnet/SqlClient/pull/1004) [#1012](https://github.com/dotnet/SqlClient/pull/1012) [#1017](https://github.com/dotnet/SqlClient/pull/1017)
+- Event source tracing improvements [#1018](https://github.com/dotnet/SqlClient/pull/1018)
+- Changes to share common files between NetFx and NetCore source code [#871](https://github.com/dotnet/SqlClient/pull/871) [#887](https://github.com/dotnet/SqlClient/pull/887)
+
+### Azure Identity dependency introduction
+**Microsoft.Data.SqlClient** now depends on **Azure.Identity** library underneath to acquire token for "Active Directory Managed Identity/MSI" and "Active Directory Service Principal" authentication modes. This changes brings below changes to public surface area:
+
+- **Breaking Change**  
+  "User Id" connection property now accepts "Client Id" instead of "Object Id" for "User-Assigned Managed Identity".  
+- **Public API**  
+  New read-only public property: `SqlAuthenticationParameters.ConnectionTimeout`
+- **Dependency**  
+  Azure.Identity v1.3.0
+
+### Event tracing improvements in SNI.dll
+`Microsoft.Data.SqlClient.SNI` (.NET Framework dependency) and `Microsoft.Data.SqlClient.SNI.runtime` (.NET Core/Standard dependency) versions have been updated to `v3.0.0-preview1.21104.2`. Event tracing in SNI.dll will no longer be enabled through a client application. Subscribing a session to the **Microsoft.Data.SqlClient.EventSource** provider through tools like xperf or perfview will be sufficient.
+
+### Enabling row version null behavior
+`SqlDataReader` returns a `DBNull` value instead of an empty `byte[]`. To active the legacy behavior, you must enable the following AppContext switch that is cached on first check for the entire process:
```

Comment:
```suggestion
`SqlDataReader` returns a `DBNull` value instead of an empty `byte[]`. To enable the legacy behavior, you must enable the following AppContext switch on application startup:
```